cornice/schemas.py

```diff
@@ -1,180 +1,78 @@
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this file,
 # You can obtain one at http://mozilla.org/MPL/2.0/.
-import webob.multidict
+from cornice.util import extract_request_data
 
-from pyramid.path import DottedNameResolver
-from cornice.util import to_list, extract_request_data
 
-
-class SchemaError(Exception):
+try:
+    import colander
+except ImportError:
     pass
-
-
-class CorniceSchema(object):
-    """Defines a cornice schema"""
-
-    def __init__(self, _colander_schema, bind_request=True):
-        self._colander_schema = _colander_schema
-        self._colander_schema_runtime = None
-        self._bind_request = bind_request
-
-    @property
-    def colander_schema(self):
-        if not self._colander_schema_runtime:
-            schema = self._colander_schema
-            schema = DottedNameResolver(__name__).maybe_resolve(schema)
-            if callable(schema):
-                schema = schema()
-            self._colander_schema_runtime = schema
-        return self._colander_schema_runtime
-
-    def bind_attributes(self, request=None):
-        schema = self.colander_schema
-        if request and self._bind_request:
-            schema = schema.bind(request=request)
-        return schema.children
-
-    def get_attributes(self, location=("body", "header", "querystring"),
-                       required=(True, False),
-                       request=None):
-        """Return a list of attributes that match the given criteria.
-
-        By default, if nothing is specified, it will return all the attributes,
-        without filtering anything.
+else:
+
+    class StrictMappingSchema(colander.MappingSchema):
+        @staticmethod
+        def schema_type():
+            return colander.Mapping(unknown='raise')
+
+    class CorniceSchema(colander.MappingSchema):
+        querystring = StrictMappingSchema()
+        headers = StrictMappingSchema()
+        body = StrictMappingSchema()
+        path = StrictMappingSchema()
```

Comment:
This PR works totally different then the past system so maybe it'd be best to actually change the name so there's no longer a CorniceSchema.  Or, CorniceSchema is some sort of compatibility class that's deprecated that helps people transition to the new system.

Any ideas for a different name?  Maybe it should be "ColanderSchema" since we're talking about adding alternate schema libraries and this class is already within the `cornice` namespace?